src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -70,15 +70,16 @@ CascadiaSettings::CascadiaSettings(const bool addDynamicProfiles)
 // Return Value:
 // - a non-ownership pointer to the profile matching the given guid, or nullptr
 //      if there is no match.
-const Profile* CascadiaSettings::FindProfile(GUID profileGuid) const noexcept
+const Profile CascadiaSettings::FindProfile(GUID profileGuid) const noexcept
 {
+    auto guid = winrt::guid(profileGuid);
     for (auto& profile : _profiles)
     {
         try
         {
-            if (profile.GetGuid() == profileGuid)
+            if (profile.HasGuid() && profile.Guid() == guid)

```

Comment:
we were having the discussion around profiles _always having guids_. Shouldn't they? Why do we need to check again?

an actual fully-loaded deserialized profile will always have a guid.